### DISCUSSION TITLE
Logging on systemd based systems

### DISCUSSION THREAD

**Discussion by ReifiedException** [netdata/netdata#12779] [Ideas]
General recommendation for applications managed by systemd is to log to stdout. Systemd takes care of capturing the output and storing it in the journal; it also handles log rotation for journal files.

I propose to change the logging policy of out applications. I dislike systemd but this is the case when it really comes to help.

**Comment by ilyam8** [comment]
> General recommendation for applications managed by systemd is to log to stdout

Can you provide a link to this recommendation?

**Reply by Ferroin** [reply]
It’s not so much a recommendation as an _expectation_.

Systemd expects services that don’t directly interact with the journal to produce log messages on stdout, which it then captures and routes with appropriate flags to the journal.

**Reply by ReifiedException** [reply]
Sure, my google still works, luckily:
https://blog.selectel.com/managing-logging-systemd/
https://news.ycombinator.com/item?id=31177567
https://linuxhint.com/understand_systemd_logs/

**Reply by ilyam8** [reply]
Ah, you mean that there are articles you can find on the Internet, I see. Asked because it wasn't clear what recommendation you are referring to.

**Reply by ReifiedException** [reply]
> Ah, you mean that there are articles you can find on the Internet, I see. Asked because it wasn't clear what recommendation you are referring to.

Are you referring to the golden tablets of commandments? Alas, it doesn't say anything about the systemd. I don't think it will enter eternity.

**Reply by ReifiedException** [reply]
> It’s not so much a recommendation as an _expectation_.
> 
> 
> 
> Systemd expects services that don’t directly interact with the journal to produce log messages on stdout, which it then captures and routes with appropriate flags to the journal.

How does this change the point? Systems that use systemd assume that applications will not log on their own. Systemd specifically provides tools to simplify logging.
If I suggested using journal, the answer would be "that would require rewriting what's already written, which is a hassle".
We already went over this when I mentioned the use of syslogd.

**Reply by rhclayto** [reply]
This would be a great option. I log as much as I can to journald via systemd.
